### PR TITLE
add text tool

### PR DESCRIPTION
**Pull request by ruirui19**
Add text as a tool to allow add text follow cursor and auto focus when adding text

### CODE REVIEW

File: src/addText.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      fontSize: this._size,
+    })
+    canvas.add(this.text);
+    canvas.setActiveObject(this.text);
+    this.text.enterEditing();
+    this.text.hiddenTextarea.focus();
+  }
+
+  doMouseUp(o) {
+    this.isDown = false;
```
Comment (janj) [inline src/addText.js:36]: what is `isDown` used for?

File: types/index.d.ts [hunk 2]
```diff
@@ -10,10 +10,12 @@ declare module 'react-sketch' {
 	    RectangleLabel: string;
 	    Select: string;
 	    Pan: string;
+		Text: string;
```
Comment (janj) [inline types/index.d.ts:13]: the spacing seems off in this file

**Review by janj — APPROVED** [review]
seems good to me

File: src/addText.js (hunk 1 above)
Comment (ruirui19) [reply src/addText.js:36]: It is to track if the drawing is finished by state

**Pull request merged**